### PR TITLE
New: Toggle between feature request modes

### PR DESCRIPTION
**Pull request by vquach2404**
This PR removes 3 samples, combining them into 1 sample. See `common-samples/issues/856` for more info.

### CODE REVIEW

File: arcgis-ios-sdk-samples/Utility network/Perform valve isolation trace/PerformValveIsolationTrace.storyboard [hunk 1]
```diff
@@ -1,9 +1,9 @@
 <?xml version="1.0" encoding="UTF-8"?>
```
Comment (yo1995) [inline arcgis-ios-sdk-samples/Utility network/Perform valve isolation trace/PerformValveIsolationTrace.storyboard:1]: This file should not be included in the PR.

File: arcgis-ios-sdk-samples/Features/Toggle between feature request modes/ToggleFeatureRequestModes.storyboard [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+                        <outlet property="statusLabel" destination="LXL-vJ-wdy" id="hak-K3-cQ4"/>
+                    </connections>
+                </viewController>
+                <placeholder placeholderIdentifier="IBFirstResponder" id="myF-4z-ytg" userLabel="First Responder" customClass="UIResponder" sceneMemberID="firstResponder"/>
+            </objects>
+            <point key="canvasLocation" x="255" y="29"/>
+        </scene>
+    </scenes>
+    <resources>
+        <systemColor name="systemBackgroundColor">
```
Comment (yo1995) [inline arcgis-ios-sdk-samples/Features/Toggle between feature request modes/ToggleFeatureRequestModes.storyboard:89]: We should use tertiary system background color.

File: arcgis-ios-sdk-samples/Features/Toggle between feature request modes/ToggleFeatureRequestModes.storyboard [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+                        <autoresizingMask key="autoresizingMask" widthSizable="YES" heightSizable="YES"/>
+                        <subviews>
+                            <view contentMode="scaleToFill" translatesAutoresizingMaskIntoConstraints="NO" id="RWQ-Y6-I3c" userLabel="Map View" customClass="AGSMapView">
+                                <rect key="frame" x="0.0" y="44" width="414" height="774"/>
+                                <color key="backgroundColor" systemColor="systemBackgroundColor"/>
+                            </view>
+                            <view contentMode="scaleToFill" translatesAutoresizingMaskIntoConstraints="NO" id="JpY-Iu-b7q">
+                                <rect key="frame" x="0.0" y="44" width="414" height="21"/>
+                                <subviews>
+                                    <label opaque="NO" userInteractionEnabled="NO" contentMode="left" horizontalHuggingPriority="251" verticalHuggingPriority="251" text="Status label" textAlignment="center" lineBreakMode="tailTruncation" baselineAdjustment="alignBaselines" adjustsFontSizeToFit="NO" translatesAutoresizingMaskIntoConstraints="NO" id="LXL-vJ-wdy">
```
Comment (yo1995) [inline arcgis-ios-sdk-samples/Features/Toggle between feature request modes/ToggleFeatureRequestModes.storyboard:27]: I feel it is better to use the custom `statusLabelBackgroundColor` we defined in the assets. Checkout samples such as `ApplyMosaicRuleToRasters` or `BufferList` and you may simply copy paste the view into this storyboard.

File: arcgis-ios-sdk-samples/Features/Toggle between feature request modes/ToggleFeatureRequestModes.storyboard [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+        <scene sceneID="xqy-uX-9SI">
+            <objects>
+                <viewController id="6uy-c7-HcL" customClass="ToggleFeatureRequestModesViewController" customModule="ArcGIS_Runtime_SDK_Samples" customModuleProvider="target" sceneMemberID="viewController">
+                    <view key="view" contentMode="scaleToFill" id="hKY-ke-PWN">
+                        <rect key="frame" x="0.0" y="0.0" width="414" height="896"/>
+                        <autoresizingMask key="autoresizingMask" widthSizable="YES" heightSizable="YES"/>
+                        <subviews>
+                            <view contentMode="scaleToFill" translatesAutoresizingMaskIntoConstraints="NO" id="RWQ-Y6-I3c" userLabel="Map View" customClass="AGSMapView">
+                                <rect key="frame" x="0.0" y="44" width="414" height="774"/>
+                                <color key="backgroundColor" systemColor="systemBackgroundColor"/>
```
Comment (yo1995) [inline arcgis-ios-sdk-samples/Features/Toggle between feature request modes/ToggleFeatureRequestModes.storyboard:22]: We use no color (default) for map view background.

File: arcgis-ios-sdk-samples.xcodeproj/project.pbxproj [hunk 5]
```diff
@@ -870,6 +870,8 @@
 		F1987D0323F7677E00DF41CC /* DisplaySubtypeSettingsViewController.swift in Sources */ = {isa = PBXBuildFile; fileRef = F1987D0223F7677E00DF41CC /* DisplaySubtypeSettingsViewController.swift */; };
```
Comment (yo1995) [inline arcgis-ios-sdk-samples.xcodeproj/project.pbxproj:870]: From the description I think the other 3 samples should be removed. But I'm not seeing these changes. So I'll probably wait for your comments rather than making changes myself. 😕

File: arcgis-ios-sdk-samples/Features/Toggle between feature request modes/ToggleFeatureRequestModesViewController.swift [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+class ToggleFeatureRequestModesViewController: UIViewController {
+    @IBOutlet weak var mapView: AGSMapView! {
+        didSet {
+            mapView.map = AGSMap(basemapStyle: .arcGISTopographic)
+            mapView.setViewpoint(AGSViewpoint(latitude: 45.5266, longitude: -122.6219, scale: 6000))
+        }
+    }
+    @IBOutlet weak var modeBarButtonItem: UIBarButtonItem!
+    @IBOutlet weak var populateBarButtonItem: UIBarButtonItem!
+    @IBOutlet weak var statusLabel: UILabel!
```
Comment (yo1995) [inline arcgis-ios-sdk-samples/Features/Toggle between feature request modes/ToggleFeatureRequestModesViewController.swift:27]:
```suggestion
    @IBOutlet var mapView: AGSMapView! {
        didSet {
            mapView.map = AGSMap(basemapStyle: .arcGISTopographic)
            mapView.setViewpoint(AGSViewpoint(latitude: 45.5266, longitude: -122.6219, scale: 6000))
        }
    }
    @IBOutlet var modeBarButtonItem: UIBarButtonItem!
    @IBOutlet var populateBarButtonItem: UIBarButtonItem!
    @IBOutlet var statusLabel: UILabel!
```

File: arcgis-ios-sdk-samples/Features/Toggle between feature request modes/ToggleFeatureRequestModesViewController.swift [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+    /// Populate for manual cache mode.
+    @IBAction func populateManualCache(_ button: UIBarButtonItem) {
+        // Set query parameters.
+        let params = AGSQueryParameters()
+        // Query for all tree conditions except "dead" with coded value '4'.
+        params.whereClause = "Condition < '4'"
+        // Show the progress HUD.
+        UIApplication.shared.showProgressHUD(message: "Populating")
+        // Populate features based on the query.
+        self.featureTable.populateFromService(with: params, clearCache: true, outFields: ["*"]) { [weak self] (result: AGSFeatureQueryResult?, error: Error?) in
```
Comment (yo1995) [inline arcgis-ios-sdk-samples/Features/Toggle between feature request modes/ToggleFeatureRequestModesViewController.swift:91]:
```suggestion
        featureTable.populateFromService(with: params, clearCache: true, outFields: ["*"]) { [weak self] (result: AGSFeatureQueryResult?, error: Error?) in
```

File: arcgis-ios-sdk-samples/Features/Toggle between feature request modes/ToggleFeatureRequestModesViewController.swift [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+    
+    /// Set the appropriate feature request mode.
+    private func changeFeatureRequestMode(to mode: AGSFeatureRequestMode) {
+        // Enable or disable the populate bar button item when appropriate.
+        if mode == .manualCache {
+            populateBarButtonItem.isEnabled = true
+        } else {
+            populateBarButtonItem.isEnabled = false
+        }
+        let map = mapView.map
```
Comment (yo1995) [inline arcgis-ios-sdk-samples/Features/Toggle between feature request modes/ToggleFeatureRequestModesViewController.swift:113]: We may just force unwrap here, to avoid optional chaining twice below.
```suggestion
        let map = mapView.map!
```

**Review by yo1995 — COMMENTED** [review]
I'll need Viv's reply to proceed, so I won't directly make changes on her branch now.

**Review by saratchandrakarumuri — APPROVED** [review]
Sample looks good now
